test/yarp/encoding_test.rb

```diff
@@ -96,5 +96,11 @@ def test_first_lexed_token
       encoding = YARP.lex("# encoding: ascii-8bit").value[0][0].value.encoding
       assert_equal Encoding.find("ascii-8bit"), encoding
     end
+
+    def test_slice_encoding
+      slice = YARP.parse("# encoding: Shift_JIS\nア").value.slice
+      assert_equal (+"ア").force_encoding(Encoding::SHIFT_JIS), slice
+      assert_equal Encoding::SHIFT_JIS, slice.encoding
```

Comment:
It would be great to test this for YARP.dump + YARP.load too (which uses serialization)

Reply:
Ah actually this is implicitly tested by the CI on JRuby & TruffleRuby which use serialization for `YARP.parse`, nevermind :)

Reply:
It does, by virtue of testing with the FFI backend